packages/engine/Source/Scene/I3SGeometry.js

```diff
@@ -382,6 +417,34 @@ I3SGeometry.prototype._generateGltf = function (
     ];
   }
 
+  const gltfMaterials = [];
+  for (let meshIndex = 0; meshIndex < meshes.length; meshIndex++) {
+    const primitives = meshes[meshIndex].primitives;
+    for (
+      let primitiveIndex = 0;
+      primitiveIndex < primitives.length;
+      primitiveIndex++
+    ) {
+      const primitive = primitives[primitiveIndex];
+      if (defined(primitive.material)) {
+        // Create as many copies of the material as specified in the mesh primitives
```

Comment:
Why are these copies needed?

Reply:
We need to adjust alphaMode parameter for each unique material that is referenced by mesh primitives, but all other material parameters should be the same for all materials.
This code is required for transparency support. The alpha channel in the vertex color attribute is applied by shaders only if BLEND mode is set for the material. However, BLEND mode affects opaque colors and makes them transparent as well. If any geometry buffer contains both opaque and transparent colors, we need to separate these vertices from each other in different mesh primitives. For each primitive we need to use separate material definition with BLEND mode for transparent colors and OPAQUE for opaque colors. We create copies of the common material definition to keep other material attributes the same.